### PR TITLE
[native] Fix crash in window queries without ORDER BY clause

### PR DESCRIPTION
**Pull request by wuxueyang96**
Presto-to-Velox plan conversion logic didn't handle the case of 
window clause without ORDER BY causing a crash.

### CODE REVIEW

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp [hunk 1]
```diff
@@ -1855,12 +1855,14 @@ VeloxQueryPlanConverter::toVeloxQueryPlan(
 
   std::vector<core::FieldAccessTypedExprPtr> sortFields;
   std::vector<core::SortOrder> sortOrders;
-  auto nodeSpecOrdering = node->specification.orderingScheme->orderBy;
-  sortFields.reserve(nodeSpecOrdering.size());
-  sortOrders.reserve(nodeSpecOrdering.size());
-  for (const auto& spec : nodeSpecOrdering) {
-    sortFields.emplace_back(exprConverter_.toVeloxExpr(spec.variable));
-    sortOrders.emplace_back(toVeloxSortOrder(spec.sortOrder));
+  if (node->specification.orderingScheme) {
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:1858]: Would you add a simple test to reproduce the original bug and ensure it doesn't get re-introduced?

CC: @aditi-pandit

**Review by mbasmanova — CHANGES_REQUESTED** [review]
@wuxueyang96 Thank you for fixing this bug.

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp (hunk 1 above)
Comment (aditi-pandit) [reply presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:1858]: @mbasmanova : Made me realize (looking at the code above) that we should test the case where there is no PARTITION BY to ensure there aren't crashes in that code path. I can take a look.

Comment (wuxueyang96) [reply presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:1858]: done

**Comment by wuxueyang96** [comment]
@mbasmanova PTAL

**Comment by mbasmanova** [comment]
@wuxueyang96 What is PTAL?

File: presto-native-execution/presto_cpp/main/types/tests/PlanConverterTest.cpp [hunk 2]
```diff
@@ -162,3 +162,7 @@ TEST_F(PlanConverterTest, scanAggBatch) {
   ASSERT_NE(partitionAndSerializeNode, nullptr);
   ASSERT_EQ(partitionAndSerializeNode->numPartitions(), 3);
 }
+
+TEST_F(PlanConverterTest, windowNodeCoredump) {
+  assertToVeloxQueryPlan("WindowNodeCoredump.json");
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/types/tests/PlanConverterTest.cpp:167]: Tests that use hard-coded JSON files are difficult to understand and maintain. Would it be possible to add a simple TPC-H-based query that manifest the issue to presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveWindowQueries.java ?

**Comment by wuxueyang96** [comment]
> @wuxueyang96 What is PTAL?

PTAL is short for `please take another look`, refer to https://en.wiktionary.org/wiki/PTAL

**Review by mbasmanova — COMMENTED** [review]
@wuxueyang96 Thank you for adding an e2e test. Let's remove the original unit test that uses hard-coded JSON file, squash commits and write a nice commit message. Check out [How to Write Better Git Commit Messages – A Step-By-Step Guide](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/) and [How to Write a Git Commit Message](https://cbea.ms/git-commit/) to learn more about how to write good commit messages.

**Review by mbasmanova — COMMENTED** [review]
@wuxueyang96 Thank you for iterating on this PR. The code looks good. The commit message needs some improvement. How about,

```
[native] Fix crash in window queries without ORDER BY clause

Presto-to-Velox plan conversion logic didn't handle the case of 
window clause without ORDER BY causing a crash.
```

[Fix for coredump which was caused by window node](https://github.com/prestodb/presto/pull/18895/commits/28f65ecf18b13791384fabb6b1e86c786c765fc6)

**Comment by wuxueyang96** [comment]
> @wuxueyang96 Thank you for iterating on this PR. The code looks good. The commit message needs some improvement. How about,
> 
> ```
> [native] Fix crash in window queries without ORDER BY clause
> 
> Presto-to-Velox plan conversion logic didn't handle the case of 
> window clause without ORDER BY causing a crash.
> ```
> 
> [Fix for coredump which was caused by window node](https://github.com/prestodb/presto/pull/18895/commits/28f65ecf18b13791384fabb6b1e86c786c765fc6)

This commit message looks better. Thanks.

**Review by mbasmanova — APPROVED** [review]
@wuxueyang96 Thank you for fixing this crash.

**Comment by mbasmanova** [comment]
@wuxueyang96 Ping me once CI is green so I can merge the PR.

**Comment by wuxueyang96** [comment]
> @wuxueyang96 Ping me once CI is green so I can merge the PR.

CI is green now.

Thank you!

**Comment by mbasmanova** [comment]
@wuxueyang96 Thank you for the contribution.

**Pull request merged**